code/src/main/webapp/JavaScript/script.js

```diff
@@ -16,13 +20,41 @@ class LoginPage {
   show() {
     this.background.classList.add("login-background");
   }
+
+  hide() {
+    // do nothing for now
+  }
 }
 
-function TransitionFromLandingToLogin() {
-  history.pushState('', 'Login', '/login.html');
-  const landingPage = new LandingPage();
-  landingPage.hide();
-  const loginPage = new LoginPage();
-  loginPage.show();
+class PageController {
+  constructor () {
+    this.landingPage = new LandingPage();
+    this.loginPage = new LoginPage();
+    this.currentlyShown = "none";
```

Comment:
Awesome! instead of using a seintinal value of "none" here, use undefined!

Otherwise, perfect!